core/src/apps/solana/ui/__init__.py

```diff
@@ -0,0 +1,281 @@
+                [
+                    (
+                        ui_property.display_name,
+                        _format_property(value, _type),
+                    )
+                ],
+            )
+        elif ui_property.account is not None:
+            account_template = instruction.get_account_template(ui_property.account)
+
+            # optional account, skip if not present
+            if ui_property.account not in instruction.parsed_accounts:
+                continue
+
+            account_value = instruction.parsed_accounts[ui_property.account]
+
+            if account_template.is_authority:
+                if signer_public_key == account_value[0]:
+                    continue
+
+            account_data = []
+            if len(account_value) == 2:
+                signer_suffix = ""
+                if account_value[0] == signer_public_key:
+                    signer_suffix = " (Signer)"
+
+                account_data.append(
+                    (
+                        ui_property.display_name,
+                        f"{base58.encode(account_value[0])}{signer_suffix}",
+                    )
+                )
+            elif len(account_value) == 3:
+                account_data.append(
+                    (f"{ui_property.display_name} is provided via a lookup table.", "")
+                )
+                account_data.append(
+                    ("Lookup table address:", base58.encode(account_value[0]))
+                )
+                account_data.append(("Account index:", f"{account_value[1]}"))
+            else:
+                raise ValueError("Invalid account value")
+
+            await confirm_properties(
+                "confirm_instruction",
+                f"{instruction_index}/{instructions_count}: {instruction.ui_name}",
+                account_data,
+            )
+        else:
+            raise ValueError("Invalid ui property")
+
+    if instruction.is_multisig:
+        await confirm_metadata(
+            "confirm_multisig",
+            "Confirm multisig",
+            "The following instruction is a multisig instruction.",
+            br_code=ButtonRequestType.Other,
+        )
+
+        assert instruction.multisig_signers is not None
+
+        signers: list[tuple[str, str]] = []
+        for i, multisig_signer in enumerate(instruction.multisig_signers, 1):
+            multisig_signer_public_key = multisig_signer[0]
+
+            path_str = ""
+            if multisig_signer_public_key == signer_public_key:
+                path_str = f" ({address_n_to_str(signer_path)})"
+
+            signers.append(
+                (f"Signer {i}{path_str}:", base58.encode(multisig_signer[0]))
+            )
+
+        await confirm_properties(
+            "confirm_instruction",
+            f"{instruction_index}/{instructions_count}: {instruction.ui_name}",
+            signers,
+        )
+
+
+def get_address_type(address_type: int) -> str:
+    if ADDRESS_SIG == address_type:
```

Comment:
It seems strange that the `address_type` is the second one in comparison (I would expect it to be first always) - it took little more time to realize this is actually `switch`

Reply:
[Changes](https://github.com/trezor/trezor-firmware/commit/41116f64b427dd88d38fd53ab9b26357c65334d7)